Assets/Project/Scripts/MenuSelectScene/LevelSelect/RoadController.cs

```diff
@@ -0,0 +1,94 @@
+﻿using UnityEngine;
+using Project.Scripts.Utils.Definitions;
+using Project.Scripts.MenuSelectScene;
+
+namespace Project.Scripts.MenuSelectScene.LevelSelect
+{
+    [RequireComponent(typeof(LineRenderer))]
+    public class RoadController : MonoBehaviour
+    {
+        /// <summary>
+        /// 開始地点の木の位置
+        /// </summary>
+        [SerializeField] private Transform _startPoint;
+
+        /// <summary>
+        /// 終了地点の木の位置
+        /// </summary>
+        [SerializeField] private Transform _endPoint;
+
+        /// <summary>
+        /// 1つ目の制御点の位置
+        /// </summary>
+        [SerializeField] private Vector2 _firstControlPoint;
+
+        /// <summary>
+        /// 2つ目の制御点の位置
+        /// </summary>
+        [SerializeField] private Vector2 _secondControlPoint;
+
+        /// <summary>
+        /// 中間地点の個数
+        /// </summary>
+        [SerializeField] [Range(0, 100)]private int _middlePointNum;
+
+        /// <summary>
+        /// 道の幅
+        /// </summary>
+        [SerializeField] [Range(0, 0.2f)] private float _width;
+
+        [SerializeField] private LineRenderer render;
+
+        private void Start()
+        {
+            // SafeAreaを考慮したContentの拡大縮小差分を制御点にも適用
+            _firstControlPoint *= ScaleContent.CONTENT_SCALE;
+            _firstControlPoint += ScaleContent.CONTENT_MARGIN;
+            _secondControlPoint *= ScaleContent.CONTENT_SCALE;
+            _secondControlPoint += ScaleContent.CONTENT_MARGIN;
+            SetPointPosition();
+        }
+
+        /// <summary>
+        /// 曲線の通過点の位置を求める
+        /// </summary>
+        public void SetPointPosition()
+        {
+            render.positionCount = _middlePointNum + 2;
+            render.startWidth = render.endWidth = (float)Screen.width * _width;
+
+            var startPointLocalPosition = _startPoint.localPosition;
+            var endPointLocalPosition = _endPoint.localPosition;
+
+            // 開始地点
+            render.SetPosition(0, startPointLocalPosition);
+
+            // 中間地点
+            for (int i = 1; i <= _middlePointNum; i++) {
+                var ratio = (float)i / (_middlePointNum + 1);
+                render.SetPosition(i, CalcCubicBezierPointPosition(startPointLocalPosition, _firstControlPoint, _secondControlPoint, endPointLocalPosition, ratio));
+            }
+
+            // 終了地点
+            render.SetPosition(_middlePointNum + 1, endPointLocalPosition);
```

Comment:
```suggestion
            // 中間地点
            for (int i = 0; i <= _middlePointNum + 1; i++) {
                var ratio = (float)i / (_middlePointNum + 1);
                render.SetPosition(i, CalcCubicBezierPointPosition(startPointLocalPosition, _firstControlPoint, _secondControlPoint, endPointLocalPosition, ratio));
            }
```

始点と終点は分けて書く必要ないかと思います。

Reply:
始点と終点は既に位置を与えられているから、改めて計算しなくても良いと思いますがどうですか？

Reply:
計算量でいうと `O(0)` なのでここは可読性を選びたいですね

Reply:
変えても良いがもう少し意見を聞きたいので、後で相談したいと思います。

Reply:
始点と終点を分けずに書きます。